keps/sig-network/3698-multi-network-requirements/README.md

```diff
@@ -0,0 +1,352 @@
+
+#### Story #5
+As a platform operator I need to connect my on-premise networks to my workload
+Pods. I need to have the ability to represent these networks in my Kubernetes
+cluster in such a way that I can easily use them in my workloads.
+
+<p align="center">
+  <img src="mn-story-5.png?raw=true" alt="multi-network story 5 On-Premise Network representation"/>
+</p>
+
+#### Story #6
+As a Kubernetes cluster operator I wish to isolate workloads based on
+namespaces and network access by overriding the Cluster Default PodNetwork for
+a given Namespace. I do not want the tenants to change their manifests for that
+purpose. Those workloads should have the same level of Kubernetes functionality:
+Services, NetworkPolicies, access to Kubernetes API. I wish to support
+“profiles” for Namespace, where I can not only change Cluster Default PodNetwork,
+but define a set of PodNetworks assigned to given Namespace that Pods created in
+that Namespace are automatically attached to.
+
+<p align="center">
+  <img src="mn-story-6.png?raw=true" alt="multi-network story 6 namespace networks profiles"/>
+</p>
+
+#### Story #7
+As a “Power User” with admin privileges I wish to have the ability to modify my
+Pod network namespace without any restrictions. I am aware that by doing this I
+might break established contracts for the kubernetes features.
+
+#### Story #8
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I need the ability to add/remove Network Interfaces to
+existing VMs without re-creating the VM. This would not be applicable for the
+Primary PodNetwork.
+
+<p align="center">
+  <img src="mn-story-8.png?raw=true" alt="multi-network story 8 network interface hot-plug"/>
+</p>
+
+#### Story #9
+As an infrastructure administrator I wish to support IPv6 SLAAC in the
+Kubernetes Pods. SLAAC's nature is to dynamically assign IP addresses and change
+them over time. Those should be reflected in Pod status. This would not be
+applicable for the Primary PodNetwork.
+
+### Requirements
+
+Below requirements are divided into phases. Each phase will produce a separate
+KEP with detailed design for specified set of requirements. Each of this KEP
+will have to take all of the requirements, from each phase, into consideration
+when created.
+
+#### Phase I (base API and reference in Pod)
+1. This effort shall not change the behavior of Today existing clusters
+2. We need an API object representing network, which a Pod, a Service or any
+other feature that wishes to support Multi-Network capability could reference
+   * We will introduce an PodNetwork API object that represents the existing 
+     infrastructure’s networking
+3. PodNetwork is decoupled from the infrastructure’s network
+definition/description created by the cluster operator. It functions as a handle
+4. PodNetwork shall not define any implementation specific parameters in its
+specification
+5. PodNetwork shall provide cluster generic options only
```

Comment:
I think there needs to be more specificity on what "cluster generic options only" means. This is in contrast to?

Reply:
This is in contrast to implementation/technology -specific parameters. Maybe "core kubernetes options only" ?